src/components/FeaturePanel/PublicTransport/requestRoutes.ts

```diff
@@ -8,17 +8,12 @@ export interface LineInformation {
 export async function requestLines(
   featureType: 'node' | 'way' | 'relation',
   id: number,
-  name: string,
-  radius = 150,
 ) {
   // use the overpass api to request the lines in
   const overpassQuery = `[out:csv(ref, colour; false; ';')];
-    ${featureType}(${id})->.center;
-    (
-      node(around.center:${radius})["public_transport"="stop_position"]["name"="${name}"];
-      nw(around.center:${radius})["highway"="bus_stop"]["name"="${name}"];
-      nwr(around.center:${radius})["amenity"="ferry_terminal"]["name"="${name}"];
-    ) -> .stops;
+  ${featureType}(${id});
+  rel(bn)["public_transport"="stop_area"];
+node(r: "stop") -> .stops;
     rel(bn.stops)["route"~"bus|train|tram|subway|light_rail|ferry|monorail"];
```

Comment:
Wow, this is some advanced query! I will have to study how this works. It could be useful on my other PR #186 .

Reply:
see also here: https://github.com/zbycz/osmapp/issues/387